core/src/blocktree_processor.rs

```diff
@@ -1230,6 +1230,57 @@ pub mod tests {
         assert_eq!(bank.last_blockhash(), blockhash);
     }
 
+    #[test]
+    fn test_process_entry_tx_random_execution_with_error() {
+        let GenesisBlockInfo {
+            genesis_block,
+            mint_keypair,
+            ..
+        } = create_genesis_block(1_000_000_000);
+        let bank = Bank::new(&genesis_block);
+
+        const NUM_TRANSFERS_PER_ENTRY: usize = 8;
+        const NUM_TRANSFERS: usize = NUM_TRANSFERS_PER_ENTRY * 32;
+        // large enough to scramble locks and results
+
+        let keypairs: Vec<_> = (0..NUM_TRANSFERS * 2).map(|_| Keypair::new()).collect();
+
+        // give everybody one lamport
+        for keypair in &keypairs {
+            bank.transfer(1, &mint_keypair, &keypair.pubkey())
+                .expect("funding failed");
+        }
+        let mut hash = bank.last_blockhash();
+
+        let entries: Vec<_> = (0..NUM_TRANSFERS)
+            .step_by(NUM_TRANSFERS_PER_ENTRY)
+            .map(|i| {
+                let mut transactions = (0..NUM_TRANSFERS_PER_ENTRY)
+                    .map(|j| {
+                        system_transaction::transfer(
+                            &keypairs[i + j],
+                            &keypairs[i + j + NUM_TRANSFERS].pubkey(),
+                            1,
+                            bank.last_blockhash(),
+                        )
+                    })
+                    .collect::<Vec<_>>();
+
+                transactions.push(system_transaction::create_account(
+                    &mint_keypair,
+                    &solana_sdk::sysvar::clock::id(), // puts a TX error in results
+                    bank.last_blockhash(),
+                    1,
+                    0,
+                    &Pubkey::new_rand(),
+                ));
+
+                next_entry_mut(&mut hash, 0, transactions)
+            })
+            .collect();
+        assert_eq!(process_entries(&bank, &entries, true), Ok(()));
```

Comment:
Won't a transaction error cause the return value of `process_entries` to be an error

Reply:
no, because this kind of transaction failure occurs in the instruction processor, ends up in the ledger